### PR TITLE
Part 2 - BlockList. remarks attached within the PR

### PR DESCRIPTION
**Pull request by yuvalmargules**

### CODE REVIEW

File: src/Lab.java [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+import java.util.HashSet;
+import java.util.Set;
 
 
 public class Lab {
-    public static void main(String[] args) throws IOException{
+    public static void main(String[] args) throws IOException {
+        Set<String> blockList = new HashSet<>();
+        if (args.length > 0)
+            blockList = launchBlockList(args[0]);
```
Comment (yuvalmargules) [inline src/Lab.java:14]: a text file with blocked domains should be loaded before receiving the query from user

File: src/Lab.java [hunk 2]
```diff
@@ -19,22 +26,42 @@ public static void main(String[] args) throws IOException{
...
-            InetAddress address = getNextServer(data);
-            senderSocket.send(new DatagramPacket(query.getData(),query.getLength(), address, DNS_PORT));
+        InetAddress userAddress = r.getAddress();
+
+        // get domain's name
+        String domainName = "";
+        byte[] data = Arrays.copyOfRange(r.getData(), 13, r.getLength());
+        for (byte cell : data) {
+            if (cell == 0) break;
+            if (cell < 20)
```
Comment (yuvalmargules) [inline src/Lab.java:38]: through debugging all ascii signs '.' where below 10. I put that number only by improvising for the correctness of the check. A better solution will be replace this.

File: src/Lab.java [hunk 3]
```diff
@@ -19,22 +26,42 @@ public static void main(String[] args) throws IOException{
...
+                domainName += '.';
+            else
+                domainName += (char) cell;
+        }
+
+        // check if the domain is listed as blocked
+        if (blockList.contains(domainName)) {
+            // TODO: return a response with an NXDOMAIN error (AKA “Name Error”, RCODE=3)
+        }
```
Comment (yuvalmargules) [inline src/Lab.java:47]: a given domain is listed in the blocklist, thus the above response should be handled.
- IMO we may create a flag of which we send to createRespone function and the function will handle between valid/invalid domains.

File: src/Lab.java [hunk 4]
```diff
@@ -44,82 +71,107 @@ public static void main(String[] args) throws IOException{
...
     }
 
     /**
      * Check response's NOERROR field
+     *
      * @param data - the payload of received packet
      * @return - true if NOERROR, else false
      */
-    public static boolean isNoError(byte[] data){
+    public static boolean isNoError(byte[] data) {
```
Comment (yuvalmargules) [inline src/Lab.java:110]: Flake8 took care of business for the sake of visibility 😂

**Review by golan13 — APPROVED** [review]

**Pull request merged**